python/spacewalk/satellite_tools/appstreams.py

```diff
@@ -0,0 +1,297 @@
+        - module_id (int): ID of the module that the APIs belong to.
+        """
+        sql = "INSERT INTO suseAppstreamApi (rpm, module_id) VALUES "
+        sql += ",".join([f"('{api}', {module_id})" for api in apis])
+        sql += " ON CONFLICT DO NOTHING"
+        rhnSQL.prepare(sql).execute()
+
+    def _insert_module_package(self, pkg: Nevra, module_id: int):
+        """
+        Inserts a modular package relation into the database.
+
+        Parameters:
+        - pkg (Nevra): Package to insert.
+        - module_id (int): ID of the module that the package belongs to.
+
+        Returns:
+        int: ID of the inserted package.
+        """
+        # Epoch '0' is not consistent in the DB. Make sure to match both '0' and NULL
+        if pkg.epoch is None or pkg.epoch == "0":
+            q_epoch_predicate = "(epoch IS NULL OR epoch = '0')"
+        else:
+            q_epoch_predicate = "epoch = :epoch"
+
+        q_insert_module_pkg = rhnSQL.prepare(
+            f"""
+            WITH package AS (
+                SELECT p.id, pn.name, pe.epoch, pe.version, pe.release, pa.label as arch
+                FROM rhnPackage p
+                    JOIN rhnPackageName pn ON p.name_id = pn.id
+                    JOIN rhnPackageEvr pe ON p.evr_id = pe.id
+                    JOIN rhnPackageArch pa ON p.package_arch_id = pa.id)
+            INSERT INTO suseAppstreamPackage (package_id, module_id)
+                SELECT p.id, :module_id FROM package p
+                 WHERE p.name = :pname
+                   AND {q_epoch_predicate}
+                   AND p.version = :version
+                   AND p.release = :release
+                   AND p.arch = :parch
+                ON CONFLICT DO NOTHING
+                RETURNING package_id
+        """
+        )
+
+        q_insert_module_pkg.execute(
+            pname=pkg.name,
+            epoch=pkg.epoch,
+            version=pkg.version,
+            release=pkg.release,
+            parch=pkg.arch,
+            module_id=module_id,
+        )
+
+        # Return the package ID if exists or None otherwise
+        row = q_insert_module_pkg.fetchone()
+        return row[0] if row else None
+
+    @staticmethod
+    def _parse_rpm_name(nevra):
+        """
+        Parses a NEVRA string into a Nevra object.
+
+        Parameters:
+        - nevra (str): NEVRA string to parse.
+
+        Returns:
+        Nevra: Nevra object containing parsed information.
+        """
+        pattern = re.compile(
+            r"(?P<name>[a-zA-Z0-9._-]+)-"
+            r"(?P<epoch>\d+:)?"
+            r"(?P<version>[a-zA-Z0-9._-]+)-"
+            r"(?P<release>[a-zA-Z0-9._+-]+)\."
+            r"(?P<arch>[a-zA-Z0-9._-]+)"
+        )
+
+        match = pattern.match(nevra)
+        if match:
+            name = match.group("name")
+            # Remove the trailing ':' if epoch is present
+            epoch = match.group("epoch")[:-1] if match.group("epoch") else None
```

Comment:
[nitpick] If you don't want to have to comment the code you can use `rstrip` instead. Either way is fine.

```suggestion
            epoch = match.group("epoch").rstrip(":") if match.group("epoch") else None
```